src/Functions/catboostEvaluate.cpp

```diff
@@ -21,66 +21,80 @@ namespace DB
 
 namespace ErrorCodes
 {
+    extern const int FILE_DOESNT_EXIST;
     extern const int ILLEGAL_TYPE_OF_ARGUMENT;
     extern const int TOO_FEW_ARGUMENTS_FOR_FUNCTION;
     extern const int ILLEGAL_COLUMN;
 }
 
-class ExternalModelsLoader;
-
-
-/// Evaluate external model.
-/// First argument - model name, the others - model arguments.
-///   * for CatBoost model - float features first, then categorical
-/// Result - Float64.
-class FunctionModelEvaluate final : public IFunction
+/// Evaluate CatBoost model.
+/// - Arguments: float features first, then categorical features.
+/// - Result: Float64.
+class FunctionCatBoostEvaluate final : public IFunction, WithContext
 {
-public:
-    static constexpr auto name = "modelEvaluate";
+private:
+    mutable std::unique_ptr<CatBoostLibraryBridgeHelper> bridge_helper;
 
-    static FunctionPtr create(ContextPtr context)
-    {
-        return std::make_shared<FunctionModelEvaluate>(context->getExternalModelsLoader());
-    }
+public:
+    static constexpr auto name = "catboostEvaluate";
 
-    explicit FunctionModelEvaluate(const ExternalModelsLoader & models_loader_)
-        : models_loader(models_loader_) {}
+    static FunctionPtr create(ContextPtr context_) { return std::make_shared<FunctionCatBoostEvaluate>(context_); }
 
+    explicit FunctionCatBoostEvaluate(ContextPtr context_) : WithContext(context_) {}
     String getName() const override { return name; }
-
     bool isVariadic() const override { return true; }
-
     bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
-
     bool isDeterministic() const override { return false; }
-
     bool useDefaultImplementationForNulls() const override { return false; }
-
     size_t getNumberOfArguments() const override { return 0; }
 
+    void initBridge(const ColumnConst * name_col) const
+    {
+        String library_path = getContext()->getConfigRef().getString("catboost_lib_path");
```

Comment:
also check `config.has("catboost_lib_path")` and throw meaningful exception? or it is checked already?

Reply:
Yeah, I just thought that we could throw a better exception message that the Poco's, but it is just a suggestion, current way is also fine.